llvm/include/llvm/IR/IntrinsicsAArch64.td

```diff
@@ -3107,6 +3107,24 @@ let TargetPrefix = "aarch64" in {
     }
   }
 
+  class SME_OuterProduct_TMOP_Intrinsic
+     : DefaultAttrsIntrinsic<[],
+         [llvm_i32_ty,
+          llvm_anyvector_ty,
+          LLVMMatchType<0>,
+          LLVMMatchType<0>,
+          llvm_nxv16i8_ty,
+          llvm_i32_ty],
+         [ImmArg<ArgIndex<0>>, ImmArg<ArgIndex<5>>,
+          IntrInaccessibleMemOnly]>;
+
+  def int_aarch64_sme_tmopa_za16 : SME_OuterProduct_TMOP_Intrinsic;
```

Comment:
nit: Maybe we should change sme_tmopa to be sme_ftmopa. It is not needed, it will make clear that is only for floating point type size

Reply:
Done.